Sources/Conduit/Networking/Data/AutoPurgingURLDataCache.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  AutoPurgingURLDataCache.swift
+//  
+//
+//  Created by Anthony Lipscomb on 8/2/21.
+//
+
+import Foundation
+
+public struct AutoPurgingURLDataCache: URLDataCache {
+
+    private let cache: NSCache<NSString, NSData>
+    private let serialQueue = DispatchQueue(
+        label: "com.mindbodyonline.Conduit.AutoPurgingDataCache-\(UUID().uuidString)"
+    )
+
+    /// Initializes an AutoPurgingURLDataCache with the desired memory capacity
+    ///
+    /// - Parameters:
+    ///     - memoryCapacity: The desired cache capacity before data eviction. Defaults to 60MB.
+    ///
+    /// - Important: The system will evict data based on different constraints within the system environment.
+    /// It is possible for the memory capacity to be surpassed and for the system to purge data at a later time.
+    public init(memoryCapacity: Int = 1_024 * 1_024 * 60) {
+        cache = NSCache()
+        cache.totalCostLimit = memoryCapacity
+    }
+
+    /// Attempts to retrieve a cached data for the given request
+    ///
+    /// - Parameters:
+    ///     - request: The request for the data
+    /// - Returns: The cached data or nil of none exists
+    public func data(for request: URLRequest) -> NSData? {
+        return _data(for: request)
+    }
+
+    private func _data(for request: URLRequest) -> NSData? {
```

Comment:
Wondering if the private functions should be removed since they both are called from public ones without any extra logic. Is there some other reason for breaking them up?